src/test/java/org/folio/readingroom/controller/PatronPermissionControllerTest.java

```diff
@@ -91,6 +97,124 @@ void testInvalidReadingRoomIdUpdatePatronPermission() throws Exception {
+  }
+
+  @Test
+  void testSuccessGetPatronPermissionWithServicePoint() throws Exception {
+    removeReadingRoomIfExists();
+    ReadingRoom readingRoom = createReadingRoom(READING_ROOM_ID_FOR_PATRON_TEST, false);
+    readingRoom.setName("TestPatronRR");
+    var servicePoints = Set.of(createServicePoint(SERVICE_POINT_ID4, SERVICE_POINT_NAME3));
+    readingRoom.servicePoints(servicePoints);
+
+    this.mockMvc.perform(
+      post("/reading-room")
+        .content(asJsonString(readingRoom))
+        .headers(defaultHeaders())
+        .contentType(MediaType.APPLICATION_JSON)
+        .accept(MediaType.APPLICATION_JSON))
+      .andExpect(status().isCreated());
+
+    PatronPermission patronPermission = createPatronPermission(UUID.randomUUID(),
+      READING_ROOM_ID_FOR_PATRON_TEST, UUID.fromString("2205005b-ca51-4a04-87fd-938eefa8f6df"));
+    this.mockMvc.perform(
+      put("/reading-room-patron-permission/2205005b-ca51-4a04-87fd-938eefa8f6df?")
+        .content(asJsonString(patronPermission))
+        .headers(defaultHeaders())
+        .contentType(MediaType.APPLICATION_JSON)
+        .accept(MediaType.APPLICATION_JSON))
+      .andExpect(status().is2xxSuccessful());
+
+    this.mockMvc.perform(
+      get("/reading-room-patron-permission/2205005b-ca51-4a04-87fd-938eefa8f6df"
+        + "?servicePointId=9865c56f-608d-4c93-94fe-f365f16e6971")
+        .headers(defaultHeaders())
+        .contentType(MediaType.APPLICATION_JSON)
+        .accept(MediaType.APPLICATION_JSON))
+      .andExpect(status().is2xxSuccessful())
+      .andExpect(jsonPath("$[0].readingRoomId").value("7a5abc9f-f3d7-4856-b8d7-6712462ca009"))
+      .andExpect(jsonPath("$.[0].userId").value("2205005b-ca51-4a04-87fd-938eefa8f6df"))
+      .andExpect(jsonPath("$.[0].access").value(ALLOWED.getValue()))
+      .andExpect(jsonPath("$", hasSize(1)));
+  }
+
+  @Test
+  void testSuccessGetPatronPermissionWithNonAssociatedServicePoint() throws Exception {
+    removeReadingRoomIfExists();
+    ReadingRoom readingRoom = createReadingRoom(READING_ROOM_ID_FOR_PATRON_TEST, false);
+    readingRoom.setName("TestPatronRR");
+
+    var servicePoints = Set.of(createServicePoint(SERVICE_POINT_ID4, SERVICE_POINT_NAME3));
+    readingRoom.servicePoints(servicePoints);
+
+    this.mockMvc.perform(
+      post("/reading-room")
+        .content(asJsonString(readingRoom))
+        .headers(defaultHeaders())
+        .contentType(MediaType.APPLICATION_JSON)
+        .accept(MediaType.APPLICATION_JSON))
+      .andExpect(status().isCreated());
+
+    this.mockMvc.perform(
+      get("/reading-room-patron-permission/2205005b-ca51-4a04-87fd-938eefa8f6df"
+        + "?servicePointId=7c5abc9f-f3d7-4856-b8d7-6712462ca008")
+        .headers(defaultHeaders())
+        .contentType(MediaType.APPLICATION_JSON)
+        .accept(MediaType.APPLICATION_JSON))
+      .andExpect(status().is2xxSuccessful())
+      .andExpect(jsonPath("$", hasSize(1)))
+      .andDo(print());
+  }
```

Comment:
We can also add another test case for reading room deleted scenario

Reply:
i think delete will not fit into Integration tests , because we don't have this field in DTOs, and we are not writing any logic for this too. Please suggest.

Reply:
Yes you are right. But we can implement in this way. We can create two reading rooms and then delete one reading room. Now, if we make a get patron permission call, it should return only one record.